Measurements/ReferenceImplementation.qs

```diff
@@ -179,10 +185,13 @@ namespace Quantum.Kata.Measurements {
 
 
     operation SuperpositionMeasurement_Reference (qs : Qubit[], bits1 : Bool[][], bits2 : Bool[][]) : Int {
+        Fact(Length(bits1[0]) == Length(bits2[0]), "Second dimension of bit Arrays should have the same length");
+        Fact(Length(bits1[0]) == Length(qs), "Second dimension of bit Arrays should be euqal to number of qubits");
+
         // measure all qubits and check in which array you can find the resulting bit string
         let meas = ResultArrayAsBoolArray (MultiM(qs));
         for (i in 0 .. Length(bits1) - 1) {
-            if (AreBitstringsEqual(bits1[i], meas)) {
+            if (AreBitstringsEqual_Reference(bits1[i], meas)) {
```

Comment:
This can be replaced by a new library function [EqualA](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.arrays.equala) (needs an `open` directive) to make the code shorter:

```suggestion
            if (EqualA(EqualB, bits1[i], meas)) {
```

Reply:
hi Mariia, i found Microsoft.Quantum.Arrays >> EqualA, but not EqualB.  if i just use EqualA in my code, i get a compile error, because the function is expecting a type:  ((('T, 'T) -> Bool), 'T[], 'T[]).  so i am not sure how to resolve this.  if there is a definition of EqualB in your code, should i see it in the compare window, here?

Reply:
It's [the library EqualB](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.logical.equalb), from M.Q.Logical namespace, that compares booleans.